### DISCUSSION TITLE
How you can set up a USB camera in Klipper

### DISCUSSION THREAD

**Discussion by ArikMerzon1** [Guilouz/Klipper-Flsun-Speeder-Pad#25] [Ideas]
My two USB cameras didn't work with Klipper.

I've found a bunch of suggestions online like 
- added the `webcam.txt` manually
- updating \ installing the `MJPG STEAMER` drivers in KIAUH  
None of the above helped

I found this library and have already tried it with 2 cameras, it works like a charm.
https://github.com/mainsail-crew/crowsnest

P.S.
Probably this repo can be added to this otherwise perfect tutorial.

**Comment by ghost** [comment]
After Crownsnest installation, my USB webcam ran perfectly again

**Comment by Guilouz** [comment]
It’s already on my repo : https://github.com/Guilouz/Klipper-Flsun-Speeder-Pad#improve-webcams-support
